packages/jbrowse-plugin-apollo/src/ApolloInternetAccount/components/ApolloLoginForm.tsx

```diff
@@ -0,0 +1,102 @@
+import {
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  TextField,
+} from '@material-ui/core'
+import React, { useState } from 'react'
+
+export const ApolloLoginForm = ({
+  authenticationURL,
+  internetAccountId,
+  handleClose,
+}: {
+  authenticationURL: string
+  internetAccountId: string
+  handleClose: (arg?: string) => void
+}) => {
+  const [username, setUsername] = useState('')
+  const [password, setPassword] = useState('')
+
+  async function onSubmit(
+    this: unknown,
+    event: React.FormEvent<HTMLFormElement>,
+  ) {
+    if (username && password) {
+      const url = new URL(authenticationURL)
+      const paramsString = `username=${username}&password=${password}`
+      const queryParams = new URLSearchParams(paramsString)
+      url.search = queryParams.toString()
+
+      fetch(url.toString(), {
+        method: 'POST',
+      })
+        .then((res) => res.json())
+        .then((data) => {
+          // If authentication was successfull then there is key 'token'
+          if ('token' in data) {
+            const responseToken = data.token
+            handleClose(responseToken)
+          } else {
+            alert('Authentication failed')
+          }
+        })
+        .catch((rejected) => {
+          alert(rejected)
+        })
```

Comment:
Generally we try to use `async/await` style promises instead of `.then()` style promises for consistency.

```suggestion
      const res = await fetch(url.toString(), {
        method: 'POST',
      })
      const data = await res.json()
      // If authentication was successfull then there is key 'token'
      if ('token' in data) {
        const responseToken = data.token
        handleClose(responseToken)
      } else {
        alert('Authentication failed')
      }
```